lib/block_view_stake.go

```diff
@@ -547,15 +547,23 @@ func (txindexMetadata *UnlockStakeTxindexMetadata) GetEncoderType() EncoderType
 // DB UTILS
 //
 
-func DBKeyForStakeByValidatorAndStaker(stakeEntry *StakeEntry) []byte {
-	data := DBKeyForStakeByValidator(stakeEntry)
-	data = append(data, stakeEntry.StakerPKID.ToBytes()...)
+func DBKeyForStakeByValidatorAndStaker(validatorPKID *PKID, stakerPKID *PKID) []byte {
+	data := DBKeyForStakeByValidator(validatorPKID)
+	data = append(data, stakerPKID.ToBytes()...)
 	return data
 }
 
-func DBKeyForStakeByValidator(stakeEntry *StakeEntry) []byte {
+func DBKeyForStakeByValidator(validatorPKID *PKID) []byte {
 	data := append([]byte{}, Prefixes.PrefixStakeByValidatorAndStaker...)
+	data = append(data, validatorPKID.ToBytes()...)
+	return data
+}
+
+func DBKeyForStakeByStakeAmount(stakeEntry *StakeEntry) []byte {
```

Comment:
We could go either way for this function. By taking in a full StakeEntry struct, this function requires the caller to have a valid struct on hand. This is fine since all of the uses of this function we have in mind already have the StakeEntry they're interested in on-hand. So the current pattern seems simpler.

The issue with the previous `DBKeyForStakeByValidatorAndStaker` function is that the intended use of the function was in call-sites where the caller only had the validatorPKID and stakerPKID on hand.

Reply:
got it - makes sense. Should we have the param be passed by value instead of by reference? So we don't have to check that stakeEntry is non-nil?

Reply:
Discussed offline. This will panic if the caller passes in nil, which is a clear signal of the unsupported nil pointer